src/backend/distributed/commands/utility_hook.c

```diff
@@ -276,6 +277,22 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 				UndistributeDisconnectedCitusLocalTables();
 			}
 			ResetConstraintDropped();
+
+			if (context == PROCESS_UTILITY_TOPLEVEL &&
+				ShouldAddNewTableToMetadata(parsetree))
+			{
+				/*
+				 * Here we need to increment command counter so that next command
+				 * can see the new table.
+				 */
+				CommandCounterIncrement();
+				CreateStmt *createTableStmt = (CreateStmt *) parsetree;
+				Oid relationId = RangeVarGetRelid(createTableStmt->relation,
+												  NoLock, false);
+				bool cascade = true;
+				bool autoConverted = false;
```

Comment:
maybe add a comment on why autoConverted is false even though we're auto-converting :)

(wondering whether we should rename that column)